src/customers/schemas/customer.schema.ts

```diff
@@ -0,0 +1,58 @@
+import { Prop, Schema, SchemaFactory } from "@nestjs/mongoose";
+import mongoose, { HydratedDocument } from "mongoose";
+import { Store } from "src/store/store.schema";
+import { UserGender } from "src/user/user.schema";
+
+export type CustomerDocument = HydratedDocument<Customer>;
+
+@Schema({
+    timestamps: true,
+})
+export class Customer {
+    @Prop({
+        required: true,
+    })
+    firstName: string;
+
+    @Prop({
+        required: true,
+    })
+    lastName: string;
+
+    @Prop({
+        required: true,
+    })
+    email: string;
+
+    @Prop({
+        required: true,
+    })
+    phoneNumber: string;
+
+    @Prop({
+        required: true,
+        type: String,
+        enum: UserGender
+    })
+    gender: string;
+
+    @Prop({
+        required: true,
+    })
+    membershipStatus: string;    
+
+    @Prop({
+        required: true,
+        unique: true
+    })
+    customersID: number;
```

Comment:
Let's use customerId. More intuitive and follows the naming pattern already used for storeId and co